### ISSUE TITLE
Tests fail randomly

### ISSUE THREAD

**Issue by joluj** [amosproj/amos-ss2021-project2-context-map#319]
Frontend tests fail randomly, this is sometimes resolved by restarting the CI Action. This need to be investigated and resolved.

**Comment by joluj** [comment]
Error seems to be an uncaught CancellationError.
```
Error:      Error: The operation was cancelled.
    CancellationError CancellationError.ts:6
    onabort           HttpService.ts:155
```

**Comment by joluj** [comment]
Fixed in PR #334, Issue #335 (multiple commits, especially ceb2cb1).
